### PR TITLE
chore: implement env vars for the demo ui

### PR DESCRIPTION
**Pull request by cre8ivejp**

### CODE REVIEW

File: pkg/auth/api/api.go [hunk 1]
```diff
@@ -432,82 +441,82 @@ func (s *authService) PrepareDemoUser() {
...
-	}
-
-	accountStorage := accountstotage.NewAccountStorage(s.mysqlClient)
-	_, err = accountStorage.GetAccountV2(
-		ctx,
-		s.config.DemoSignInConfig.Email,
-		s.config.DemoSignInConfig.OrganizationId,
-	)
-	if err != nil && errors.Is(err, accountstotage.ErrAccountNotFound) {
-		err = accountStorage.CreateAccountV2(ctx, &domain.AccountV2{
```
Comment (cre8ivejp) [inline pkg/auth/api/api.go:495]: I moved this to be executed in the MySQL transaction.

File: pkg/auth/api/api.go [hunk 2]
```diff
@@ -432,82 +441,82 @@ func (s *authService) PrepareDemoUser() {
...
+					Name:             "Demo Account",
+					OrganizationRole: acproto.AccountV2_Role_Organization_ADMIN,
+					EnvironmentRoles: []*acproto.AccountV2_EnvironmentRole{
+						{
+							EnvironmentId: config.EnvironmentId,
+							Role:          acproto.AccountV2_Role_Environment_EDITOR,
+						},
 					},
+					CreatedAt: now.Unix(),
+					UpdatedAt: now.Unix(),
```
Comment (cre8ivejp) [inline pkg/auth/api/api.go:513]: I added the values that were missing.

File: manifests/bucketeer/charts/backend/templates/env-js-configmap.yaml [hunk 3]
```diff
@@ -10,4 +10,4 @@ metadata:
     heritage: {{ .Release.Service }}
 data:
   env.js: |-
-    window.env = {"GOOGLE_TAG_MANAGER_ID":"{{ .Values.env.googleTagManagerId }}"};
+    window.env = {"GOOGLE_TAG_MANAGER_ID":"{{ .Values.env.googleTagManagerId }}", "DEMO_SIGN_IN_ENABLED":{{ .Values.oauth.demoSignIn.enabled }}, "DEMO_SIGN_IN_EMAIL":"{{ .Values.oauth.demoSignIn.email }}", "DEMO_SIGN_IN_PASSWORD":"{{ .Values.oauth.demoSignIn.password }}"};
```
Comment (cre8ivejp) [inline manifests/bucketeer/charts/backend/templates/env-js-configmap.yaml:13]: All these 3 env vars will be used on the sign-in UI page.

File: pkg/auth/api/signin.go [hunk 4]
```diff
@@ -34,8 +34,9 @@ func (s *authService) SignIn(
 	if err != nil {
 		return nil, err
 	}
-	if request.Email != s.config.DemoSignInConfig.Email &&
```
Comment (cre8ivejp) [inline pkg/auth/api/signin.go:37]: I fixed the condition because if one of the two was correct, it authorized and generated the token.

**Review by Ubisoft-potato — APPROVED** [review]
LGTM!

**Pull request merged**